spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/it/DatastoreIntegrationTests.java

```diff
@@ -536,18 +538,43 @@ public void referenceTest() {
 	}
 
 	@Test
-	public void lazyReferenceTest() {
+	public void lazyReferenceCollectionTest() {
 		ReferenceEntry parent = saveEntitiesGraph();
 
-		//Saving an entity with not loaded lazy field
 		ReferenceEntry lazyParent = this.datastoreTemplate.findById(parent.id, ReferenceEntry.class);
 
+		//Saving an entity with not loaded lazy field
 		this.datastoreTemplate.save(lazyParent);
 
 		ReferenceEntry loadedParent = this.datastoreTemplate.findById(lazyParent.id, ReferenceEntry.class);
 		assertThat(loadedParent.children).containsExactlyInAnyOrder(parent.children.toArray(new ReferenceEntry[0]));
 	}
 
+
+	@Test
+	public void lazyReferenceTest() throws InterruptedException {
+		LazyEntity lazyParentEntity = new LazyEntity(new LazyEntity(new LazyEntity()));
+		this.datastoreTemplate.save(lazyParentEntity);
+
+		LazyEntity loadedParent = this.datastoreTemplate.findById(lazyParentEntity.id, LazyEntity.class);
+
+		//Saving an entity with not loaded lazy field
+		this.datastoreTemplate.save(loadedParent);
+
+		loadedParent = this.datastoreTemplate.findById(loadedParent.id, LazyEntity.class);
```

Comment:
does this have the same problem that we had in previous PR where the equality does not necessarily prove anything?

Reply:
it did prove everything we needed actually :) I only changed it for readability. 
But in this case I'm testing multi-level lazy loading.